classes/userGroup/UserGroup.php

```diff
@@ -3,263 +3,387 @@
+
+    /**
+     * The primary key associated with the table.
+     *
+     * @var string
+     */
+    protected $primaryKey = 'user_group_id';
+
+    /**
+     * Indicates if the model should be timestamped.
+     *
+     * @var bool
+     */
+    public $timestamps = false;
+
     /**
-     * Get the role ID
+     * The attributes that are mass assignable.
      *
-     * @return int ROLE_ID_...
+     * @var array
      */
-    public function getRoleId()
+    protected $fillable = [
+        'context_id',
+        'role_id',
+        'is_default',
+        'show_title',
+        'permit_self_registration',
+        'permit_metadata_edit',
+        'masthead',
+    ];
+
+    /**
+     * The attributes that should be cast to native types.
+     *
+     * @var array
+     */
+    protected function casts(): array
     {
-        return $this->getData('roleId');
+        return [
+            'context_id' => 'integer',
+            'role_id' => 'integer',
+            'is_default' => 'boolean',
+            'show_title' => 'boolean',
+            'permit_self_registration' => 'boolean',
+            'permit_metadata_edit' => 'boolean',
+            'masthead' => 'boolean',
+            // multilingual attributes will be handled through accessors and mutators
+        ];
     }
 
     /**
-     * Set the role ID
+     * List of attributes associated with the settings
      *
-     * @param int $roleId ROLE_ID_...
+     * @var array
      */
-    public function setRoleId($roleId)
+    protected array $settings = [
+        'name',
+        'abbrev',
+        'nameLocaleKey',
+        'abbrevLocaleKey',
+    ];
+
+    /**
+     * The list of multilingual attributes.
+     *
+     * @var array
+     */
+    protected array $multilingualProps = [
+        'name',
+        'abbrev',
+    ];
+
+    /**
+     * Get the settings table name
+     */
+    public function getSettingsTable(): string
     {
-        $this->setData('roleId', $roleId);
+        return 'user_group_settings';
     }
 
     /**
-     * Get the role path
-     *
-     * @return string Role path
+     * Get the schema name for the model
      */
-    public function getPath()
+    public static function getSchemaName(): ?string
     {
-        return $this->getData('path');
+        return 'userGroup';
     }
 
     /**
-     * Set the role path
-     * $param $path string
+     * Ensure casts are string values.
+     *
+     * @param array $casts
+     * @return array
      */
-    public function setPath($path)
+    protected function ensureCastsAreStringValues($casts): array
     {
-        $this->setData('path', $path);
+        return array_map(fn($cast) => (string) $cast, $casts);
     }
 
     /**
-     * Get the context ID
+     * setAttribute method for multilingual attributes
+     *
+     * @param string $key
+     * @param mixed $value
+     * @return $this
      */
-    public function getContextId(): ?int
+    public function setAttribute($key, $value)
```

Comment:
I don't think we need to override setAttribute() here in order to allow to set multilingual values as usual strings. It's better to think about general solution